### PR TITLE
testutil: Moving all e2e utils to separate package, added TB union.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/thanos/main_test.go [hunk 1]
```diff
@@ -11,6 +11,7 @@ import (
 	"path/filepath"
 
 	"github.com/thanos-io/thanos/pkg/block/metadata"
+	"github.com/thanos-io/thanos/pkg/testutil/e2eutil"
```
Comment (kakkoyun) [inline cmd/thanos/main_test.go]: nit: Shall we group the imports? Does it even matter for the tests?

File: pkg/testutil/testorbench.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+func (tb *UnionTB) ResetTimer() {
+	if b, ok := tb.TB.(*testing.B); ok {
+		b.ResetTimer()
+		return
+	}
+}
+
+func (tb *UnionTB) IsBenchmark() bool {
+	if _, ok := tb.TB.(*testing.B); ok {
```
Comment (kakkoyun) [inline pkg/testutil/testorbench.go]: I'm totally sure if this makes easier to read or not, but the following could also be used
```suggestion
	_, ok := tb.TB.(*testing.B)
	return ok
```

File: cmd/thanos/main_test.go (hunk 1 above)
Comment (bwplotka) [reply cmd/thanos/main_test.go]: we should but we don't have tools for that so it's tedious and I am lazy (: 

@jojohappy if you were asking if still valid: https://github.com/thanos-io/thanos/issues/984

File: pkg/testutil/testorbench.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	if b, ok := tb.TB.(*testing.B); ok {
+		return b.N
+	}
+	return 1
+}
+
+func (tb *UnionTB) ResetTimer() {
+	if b, ok := tb.TB.(*testing.B); ok {
+		b.ResetTimer()
+		return
```
Comment (GiedriusS) [inline pkg/testutil/testorbench.go]: very minor nit: not needed return :P

File: pkg/testutil/testorbench.go (hunk 2 above)
Comment (bwplotka) [reply pkg/testutil/testorbench.go]: Makes sense :+1:

**Pull request merged**